java/core/src/main/java/com/microsoft/bond/StructBondType.java

```diff
@@ -0,0 +1,1359 @@
+        return this.baseStructType;
+    }
+
+    /**
+     * Gets an array containing the descriptors of the struct fields.
+     *
+     * @return an array containing the descriptors of the struct fields
+     */
+    public final StructField<?>[] geStructFields() {
+        return this.structFields.clone();
+    }
+
+    /**
+     * Builds a new {@link SchemaDef} instance describing the schema of this struct.
+     *
+     * @return a new schema definition instance
+     */
+    public final SchemaDef buildSchemaDef() {
+        SchemaDef schemaDef = new SchemaDef();
+        HashMap<StructBondType<?>, StructDefOrdinalTuple> typeDefMap =
+                new HashMap<StructBondType<?>, StructDefOrdinalTuple>();
+        schemaDef.root = this.createSchemaTypeDef(typeDefMap);
+        StructDef[] tempArray = new StructDef[typeDefMap.size()];
+        for (Map.Entry<StructBondType<?>, StructDefOrdinalTuple> e : typeDefMap.entrySet()) {
+            StructDefOrdinalTuple structDefInfo = e.getValue();
+            tempArray[structDefInfo.ordinal] = structDefInfo.structDef;
+        }
+        schemaDef.structs.addAll(Arrays.asList(tempArray));
+        return schemaDef;
+    }
+
+    @Override
+    public final String getName() {
+        // rely on generated class
+        return this.getValueClass().getSimpleName();
+    }
+
+    @Override
+    public final String getQualifiedName() {
+        // rely on generated class
+        return this.getValueClass().getName();
+    }
+
+    @Override
+    public final BondDataType getBondDataType() {
+        return BondDataType.BT_STRUCT;
+    }
+
+    @Override
+    public final boolean isNullableType() {
+        return false;
+    }
+
+    @Override
+    public final boolean isGenericType() {
+        return this.genericTypeSpecialization != null;
+    }
+
+    @Override
+    public final BondType<?>[] getGenericTypeArguments() {
+        return this.genericTypeSpecialization != null ?
+                this.genericTypeSpecialization.genericTypeArguments.clone() : null;
+    }
+
+    @Override
+    public final Class<TStruct> getPrimitiveValueClass() {
+        return null;
+    }
+
+    @Override
+    protected final TStruct newDefaultValue() {
+        return this.newInstance();
+    }
+
+    /**
+     * Instantiates a new instance of this struct type.
+     *
+     * @return new struct instance
+     */
+    public abstract TStruct newInstance();
+
+    @Override
+    protected final void serializeValue(SerializationContext context, TStruct value) throws IOException {
+        this.verifyNonNullableValueIsNotSetToNull(value);
+        context.writer.writeStructBegin(this);
+        if (this.baseStructType != null) {
+            this.baseStructType.serializeValueHelperForBase(context, value);
+        }
+        this.serializeStructFields(context, value);
+        context.writer.writeStructEnd();
+    }
+
+    // recursive helper
+    private void serializeValueHelperForBase(
```

Comment:
Sure, renamed.

---
In reply to: [123397272](https://github.com/Microsoft/bond/pull/502#discussion_r123397272) [](ancestors = 123397272)